File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 6/9/2025, 1:57:04 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int,int> bruh;
        for(int i=0;i<n;i++)
        {
            int complement = target - nums[i];

            if(bruh.find(complement)!=bruh.end())
            return {bruh[complement], i};
        bruh[nums[i]] = i;
        }
    return {};
    }
};

File: 1-TwoSum/1-TwoSum.js
// Last updated: 6/9/2025, 1:57:02 PM
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let seen = new Map();
    for(let i=0;i<nums.length;i++)
    {
        const num = nums[i];
        if( seen.has(num))
        return [i, seen.get(num)];
        seen.set (target-num, i );

    }
    return[]
};